worlds/smw/Locations.py

```diff
@@ -234,17 +856,146 @@ def __init__(self, player: int, name: str = '', address: int = None, parent=None
+    LocationName.special_zone_6_coin_block_23,
+    LocationName.special_zone_6_coin_block_24,
+    LocationName.special_zone_6_coin_block_25,
+    LocationName.special_zone_6_coin_block_26,
+    LocationName.special_zone_6_coin_block_27,
+    LocationName.special_zone_6_coin_block_28,
+    LocationName.special_zone_6_powerup_block_2,
+    LocationName.special_zone_6_coin_block_29,
+    LocationName.special_zone_6_coin_block_30,
+    LocationName.special_zone_6_coin_block_31,
+    LocationName.special_zone_6_coin_block_32,
+    LocationName.special_zone_6_coin_block_33,
+    LocationName.special_zone_5_yoshi_block_1,
+    LocationName.special_zone_1_vine_block_1,
+    LocationName.special_zone_1_vine_block_2,
+    LocationName.special_zone_1_vine_block_3,
+    LocationName.special_zone_1_vine_block_4,
+    LocationName.special_zone_1_life_block_1,
+    LocationName.special_zone_1_vine_block_5,
+    LocationName.special_zone_1_blue_pow_block_1,
+    LocationName.special_zone_1_vine_block_6,
+    LocationName.special_zone_1_powerup_block_1,
+    LocationName.special_zone_1_pswitch_coin_block_1,
+    LocationName.special_zone_1_pswitch_coin_block_2,
+    LocationName.special_zone_1_pswitch_coin_block_3,
+    LocationName.special_zone_1_pswitch_coin_block_4,
+    LocationName.special_zone_1_pswitch_coin_block_5,
+    LocationName.special_zone_1_pswitch_coin_block_6,
+    LocationName.special_zone_1_pswitch_coin_block_7,
+    LocationName.special_zone_1_pswitch_coin_block_8,
+    LocationName.special_zone_1_pswitch_coin_block_9,
+    LocationName.special_zone_1_pswitch_coin_block_10,
+    LocationName.special_zone_1_pswitch_coin_block_11,
+    LocationName.special_zone_1_pswitch_coin_block_12,
+    LocationName.special_zone_1_pswitch_coin_block_13,
+    LocationName.special_zone_2_powerup_block_1,
+    LocationName.special_zone_2_coin_block_1,
+    LocationName.special_zone_2_coin_block_2,
+    LocationName.special_zone_2_powerup_block_2,
+    LocationName.special_zone_2_coin_block_3,
+    LocationName.special_zone_2_coin_block_4,
+    LocationName.special_zone_2_powerup_block_3,
+    LocationName.special_zone_2_multi_coin_block_1,
+    LocationName.special_zone_2_coin_block_5,
+    LocationName.special_zone_2_coin_block_6,
+    LocationName.special_zone_3_powerup_block_1,
+    LocationName.special_zone_3_yoshi_block_1,
+    LocationName.special_zone_3_wings_block_1,
+    LocationName.special_zone_4_powerup_block_1,
+    LocationName.special_zone_4_star_block_1
+]
+
 location_table = {}
 
 
-def setup_locations(world, player: int):
+def setup_locations(world: World):
     location_table = {**level_location_table}
 
-    # Dragon Coins here
-    if world.dragon_coin_checks[player].value:
+    if world.options.dragon_coin_checks:
         location_table.update({**dragon_coin_location_table})
 
-    if world.goal[player] == "yoshi_egg_hunt":
+    if world.options.moon_checks:
+        location_table.update({**moon_location_table})
```

Comment:
here you are unpacking, then repacking, then updating. You can skip the first two.
```suggestion
        location_table.update(moon_location_table)
```